core/src/main/java/org/springframework/security/authorization/AuthorizationObservationContext.java

```diff
@@ -73,15 +73,32 @@ public T getObject() {
 	 * @return the observed {@link AuthorizationDecision}
 	 */
 	public AuthorizationDecision getDecision() {
-		return this.decision;
+		return (AuthorizationDecision) this.authorizationResult;
```

Comment:
Please perform a type check here. The reason is so that the application can give a more informative error than `ClassCastException`. Maybe something like:

```java
Assert.isInstanceOf(AuthorizationDecision.class, "please call getAuthorizationResult instead. If you must call getDecision, please ensure that the result you provide is of type AuthorizationDecision");
return (AuthorizationDecision) this.authorizationResult;
```